lib/client-js/maps.js

```diff
@@ -1,124 +1,211 @@
+
+//             if (data.map !== '') {
+//                 break;
+//             }
+//         } else if (building.toUpperCase() === campus.buildings[i].id.toUpperCase()) {
+//             data.building = campus.buildings[i].id;
+
+//             if (room !== null && pData.room !== undefined && pData.room !== '') {
+
+//                 for (var j = 0; j < campus.buildings[i].floors.length; j++) {
+
+//                     if (room[campus.buildings[i].floorIndex].toUpperCase() === campus.buildings[i].floors[j].id.toUpperCase() || floor.toUpperCase() === campus.buildings[i].floors[j].id.toUpperCase()) {
+//                         data.floor = campus.buildings[i].floors[j].id;
+//                         data.map = campus.buildings[i].dir + campus.buildings[i].floors[j].map;
+//                         data.room = room;
+//                         break;
+//                     }
+//                 }
+//             } else {
+//                 data.floor = campus.buildings[i].floors[0].id;
+//                 data.map = campus.buildings[i].dir + campus.buildings[i].floors[0].map;
+//                 data.room = room;
+//                 break;
+//             }
+//         }
+//     }
+
+//     debug.msg('Building= ' + data.building);
+//     debug.msg('Floor= ' + data.floor);
+//     debug.msg('Map= ' + data.map);
+//     debug.msg('Room= ' + data.room);
+
+//     if (data.roomName !== '' && data.map === '') {
+//         alert('Were you trying to search by nickname? Unfortunately, "' +
+//             name + '" is not recognized.\nPlease check your spelling, ' +
+//             'or try a more common name.\n\nExample: Search for "Moes" not "Mo"');
+//         data.building = 'Campus';
+//         data.map = campus.map;
+//     } else if (data.map === '') {
+//         debug.error("Did you mean to search for \"" + pData.builing + ' ' + pData.room + "\"?\nThat building doesn't exist.");
+//         alert("Did you mean to search for building " + pData.building + "?\nThat building doesn't exist.");
+//         data.building = 'Campus';
+//         data.map = campus.map;
+//     }
+//     debug.end();
+//     return data;
+// }
+//Refactored, sinmplified, and cleaned up code
 function switchMap(search) {
     debug.group('Switch Map');
-    var data = {
-        map: '',
-        building: '',
-        floor: '',
-        room: '',
-        roomName: ''
+    const data = {
+      map: '',
+      building: '',
+      floor: '',
+      room: '',
+      roomName: ''
     };
-
-    var pData = parseSearch(search);
-    var building = pData.building;
-    var floor = '';
-    var room = pData.room;
-    data.roomName = pData.roomName;
-    debug.msg('RoomName "' + data.roomName + '"');
-
-
+  
+    const { building, room, roomName } = parseSearch(search);
+    data.roomName = roomName;
+    debug.msg(`RoomName "${data.roomName}"`);
+  
+    const matchingBuilding = campus.buildings.find(b => b.id.toUpperCase() === building.toUpperCase());
```

Comment:
More of a preference thing than anything else, but I would normally use `toLowerCase` either works.